src/azure-cli/azure/cli/command_modules/keyvault/custom.py

```diff
@@ -334,6 +334,11 @@ def create_keyvault(cmd, client,  # pylint: disable=too-many-locals
     from azure.graphrbac.models import GraphErrorException
     from azure.graphrbac import GraphRbacManagementClient
 
+    try:
+        return client.get(resource_group_name=resource_group_name, vault_name=vault_name)
```

Comment:
if get, will directly return existing keyvault without updating? This might be problem when user creating params is different with existing properties. I think 2 options here
- a nice message. "The resource exists already"
- get firstly, if exists, do put to update

Reply:
Portal behavior is to disallow the creation if the vault name already exists. Shall we follow it?

Reply:
Hi @yungezz , @fengzhou-msft Thanks for the good suggestion.
It makes more sense to just disallow the creation, and prompt some messages as well.
Update the resource may cause breaking changes when handling default values. It's better to use `az keyvault update` command to explicitly do the update.